test/utils/update_resources.go

```diff
@@ -33,11 +36,47 @@ const (
 	waitRetryTImeout    = 5 * time.Minute
 )
 
-func ScaleResourceWithRetries(scaler kubectl.Scaler, namespace, name string, size uint, gr schema.GroupResource) error {
-	waitForScale := kubectl.NewRetryParams(updateRetryInterval, updateRetryTimeout)
-	waitForReplicas := kubectl.NewRetryParams(waitRetryInterval, waitRetryTImeout)
-	if err := scaler.Scale(namespace, name, size, nil, waitForScale, waitForReplicas, gr); err != nil {
+func RetryErrorCondition(condition wait.ConditionFunc) wait.ConditionFunc {
+	return func() (bool, error) {
+		done, err := condition()
+		if err != nil {
+			return false, nil
+		}
+		return done, nil
+	}
+}
+
+func ScaleResourceWithRetries(scalesGetter scaleclient.ScalesGetter, namespace, name string, size uint, gr schema.GroupResource) error {
+	scaler := kubectl.NewScaler(scalesGetter)
+	preconditions := &kubectl.ScalePrecondition{-1, ""}
+	cond := RetryErrorCondition(kubectl.ScaleCondition(scaler, preconditions, namespace, name, size, nil, gr))
+	err := wait.PollImmediate(updateRetryInterval, updateRetryTimeout, cond)
+	if err == nil {
+		err = wait.PollImmediate(
+			waitRetryInterval,
+			waitRetryTImeout,
+			scaleHasDesiredReplicas(scalesGetter, gr, name, namespace, int32(size)))
+		if err == wait.ErrWaitTimeout {
+			err = fmt.Errorf("timed out waiting for %q to be synced", name)
+		}
+	}
+	if err != nil {
 		return fmt.Errorf("Error while scaling %s to %d replicas: %v", name, size, err)
 	}
 	return nil
 }
+
+func scaleHasDesiredReplicas(sClient scaleclient.ScalesGetter, gr schema.GroupResource, resourceName string, namespace string, desiredReplicas int32) wait.ConditionFunc {
```

Comment:
Let's make scaleHasDesiredReplicas in pkg/kubectl/scale.go public method instead of copying it here.

Reply:
done